### DISCUSSION TITLE
如何测试并计算Controller模式的RTO和RPO？

### DISCUSSION THREAD

**Discussion by onejimmyboy** [apache/rocketmq#7376] [Q&A]
如何测试并计算Controller模式的RTO和RPO呐？
RTO：可以用发送消息异常到发送消息正常直接的时间间隔标识
RPO怎么表示那？

**Comment by RongtongJin** [comment]
> 如何测试并计算Controller模式的RTO和RPO呐？ RTO：可以用发送消息异常到发送消息正常直接的时间间隔标识 RPO怎么表示那？

可以衡量切换前后数据是否丢失

**Reply by onejimmyboy** [reply]
谢谢回复，
如何进一步测试呐？RPO是否可以理解为数据恢复的时间，是否可以用消费者来测试，启动消费者从消费异常，到消费正常这段时间表示RPO呐？

**Reply by RongtongJin** [reply]
RPO标志系统能够容忍的最大数据丢失量。所以可以检测消息是否有丢失

**Reply by onejimmyboy** [reply]
检测切换前后消息是否有丢失？这个怎么检测呀！能给一个思路吗？切换前先预制一部分消息，然后看切换后看这些消息还在不在？

**Reply by RongtongJin** [reply]
可以参考一下这个https://github.com/openmessaging/openchaos/blob/master/chaos-framework/src/main/java/io/openchaos/checker/QueueChecker.java
